### PR TITLE
Add a check for ALLOW_UNFILTERED_UPLOADS

### LINKED ISSUE #8: Check: NOT PERMITTED: Use of ALLOW_UNFILTERED_UPLOADS

### PR DESCRIPTION
**Pull request by EvanHerman**
Resolves #8 

Check for the existence of `ALLOW_UNFILTERED_UPLOADS` in the codebase.

### CODE REVIEW

**Review by dd32 — APPROVED** [review]
LGTM.

I would have thought this would be covered by the `WordPress.WP.DiscouragedConstants` PHPCS check, but evidently not:

https://github.com/WordPress/WordPress-Coding-Standards/blob/cb90f0181b420776c3fa0ea1925f6746b5d1823f/WordPress/Sniffs/WP/DiscouragedConstantsSniff.php#L28

File: checks/check-constants.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?php
+namespace WordPressdotorg\Plugin_Check\Checks;
+use WordPressdotorg\Plugin_Check\{Error, Guideline_Violation, Message, Notice, Warning};
+
+class Code_Constants extends Check_Base {
+
+	function check_allow_unfiltered_uploads() {
```
Comment (bordoni) [inline checks/check-constants.php]:
```suggestion
	public function check_allow_unfiltered_uploads() {
```

File: checks/check-constants.php [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+		if (
+			$this->scan_matching_files_for_needle( 'ALLOW_UNFILTERED_UPLOADS', '\.php$' )
+		) {
+			return new Error(
+				'allow_unfiltered_uploads_detected',
+				__( 'ALLOW_UNFILTERED_UPLOADS is not permitted.', 'wporg-plugins' )
+			);
+		}
```
Comment (bordoni) [inline checks/check-constants.php]: Invert the conditional and bail to reduce cognitive load for long-term maintenance.

```suggestion
		if ( ! $this->scan_matching_files_for_needle( 'ALLOW_UNFILTERED_UPLOADS', '\.php$' ) ) {
			return;
		}

		return new Error(
			'allow_unfiltered_uploads_detected',
			__( 'ALLOW_UNFILTERED_UPLOADS is not permitted.', 'wporg-plugins' )
		);
```

**Review by bordoni — CHANGES_REQUESTED** [review]
Small changes.

**Review by bordoni — APPROVED** [review]

**Pull request merged**